numpy/core/tests/test_umath.py

```diff
@@ -789,6 +789,32 @@ def assert_complex_equal(x, y):
                 assert_complex_equal(np.power(zero, -p), cnan)
             assert_complex_equal(np.power(zero, -1+0.2j), cnan)
 
+    # Testing 0^{Non-zero} issue 18378
+    def test_zero_power_nonzero(self):
+        zero = np.array([0.0j])
+        cnan = np.array([complex(np.nan, np.nan)])
+
+        def assert_complex_equal(x, y):
+            assert_array_equal(x.real, y.real)
+            assert_array_equal(x.imag, y.imag)
+
+        #Complex powers with positive real part will not generate a warning
+        assert_complex_equal(np.power(zero, 1+4j), zero)
+        assert_complex_equal(np.power(zero, 2-3j), zero)
+        #Testing zero values when real part is greater than zero
+        assert_complex_equal(np.power(zero, 1+1j), -zero)
+        assert_complex_equal(np.power(zero, 1+0j), -zero)
```

Comment:
```suggestion
        assert_complex_equal(np.power(zero, 1+1j), zero)
        assert_complex_equal(np.power(zero, 1+0j), zero)
```

Reply:
@scratchmex Thank you for the comments. If maintainers agree to make these changes then I can add another commit applying those changes. 😄

Reply:
@seberg ping

Reply:
@prithvitewatia I committed some of the changes, others are waiting for responses. Please refactor or comment.